A103-xds-composite-filter.md

```diff
@@ -0,0 +1,171 @@
+          This field must be present. The configured value will be capped at
+          100%.
+        - runtime_key: This field will be ignored, since gRPC does not
+          have a runtime system.
+- matcher: gRPC will not support this deprecated field.
+
+We will also support per-route overrides via the
+[`envoy.extensions.common.matching.v3.ExtensionWithMatcherPerRoute`
+proto](https://github.com/envoyproxy/envoy/blob/0685d7bf568485eb112df2a9c73248cb8bfc1c37/api/envoy/extensions/common/matching/v3/extension_matcher.proto#L39C9-L39C37).
+In this proto, the
+[xds_matcher](https://github.com/envoyproxy/envoy/blob/0685d7bf568485eb112df2a9c73248cb8bfc1c37/api/envoy/extensions/common/matching/v3/extension_matcher.proto#L41)
+must be validated the same way as the corresponding field in the
+top-level config.  The value of this field will replace the value of the
+field in the top-level config.
+
+The parsed representation of the composite filter's config will be a
+matcher tree, using the same unified matcher API implemented for [A77].
+The actions in the matcher tree will be one of two possible values: a
+parsed filter config, or an indication that the filter should be skipped
+(in the case of a `SkipFilter` proto).
+
+Note that in order to avoid potential stack overflows, we will impose
+a maximum recursion depth of 8 when parsing HTTP filter configs.
+
+### CEL Attributes
+
+In addition to the CEL request attributes described in [A77], we will
+also add support for some additional [CEL
+attributes](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/attributes.html#configuration-attributes)
+that we expect to be useful for the composite filter.
+
+On both the gRPC client and server sides, we will add support for the
+`xds.route_metadata.filter_metadata` attribute.  To support this, we will
+add support for parsing the [`Route.metadata`
+field](https://github.com/envoyproxy/envoy/blob/f384ab2b3e3aa0564ef25f57dc2ed8ad61eaf0cb/api/envoy/config/route/v3/route_components.proto#L319)
+in the xDS RouteConfiguration.  This field will be validated the same
+way as cluster metadata, as described in [A83].  The parsed metadata map
+will be added to the route in the parsed RouteConfiguration resource,
+and that map will be accessed by this CEL attribute.  Note that we will
+support only `filter_metadata`, not `typed_filter_metadata`, so that we
+do not have to handle protobuf descriptor functionality; to that end, we
+will use only those entries in the parsed metadata map that correspond
+to `google.protobuf.Struct` type.
+
+We will also add support for the following attributes on the gRPC server
+side only (these attributes are not relevant on the client side):
+- `source.ip`
+- `source.port`
+- `connection.requested_server_name`
+- `connection.tls_version`
+- `connection.sha256_peer_certificate_digest`
+
+### Filter Behavior
+
+Because all of the CEL attributes that we are exposing are available
```

Comment:
I need to think about this more in Java. This is harder for us than just processing it in the filter itself. I'll need to think about how we'd want to implement the approach mentioned here.